src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/ReadsPipelineSpark.java

```diff
@@ -105,7 +106,14 @@ protected void runTool(final JavaSparkContext ctx) {
         }
 
         JavaRDD<GATKRead> initialReads = getReads();
-        JavaRDD<GATKRead> markedReads = MarkDuplicatesSpark.mark(initialReads, getHeaderForReads(), duplicatesScoringStrategy, new OpticalDuplicateFinder());
+        JavaRDD<GATKRead> markedReadsWithOD = MarkDuplicatesSpark.mark(initialReads, getHeaderForReads(), duplicatesScoringStrategy, new OpticalDuplicateFinder());
+
+        //The OD attribute was added to each read for optical dups.
+        // Now we have to clear it to avoid polluting the output.
+        final JavaRDD<GATKRead> markedReads = markedReadsWithOD.map(read -> {
+            read.clearAttribute(MarkDuplicatesSparkUtils.OPTICAL_DUPLICATE_TOTAL_ATTRIBUTE_NAME);
+            return read;
+        });
```

Comment:
oh, that's because MarkDuplicatesSpark.runTool prints the metrics from the reads. I agree it's bogus. https://github.com/broadinstitute/gatk/issues/1244

I now extracted the cleanup to its own method